pkg/operator/ceph/object/admin.go

```diff
@@ -212,13 +212,10 @@ func RunAdminCommandNoMultisite(c *Context, expectJSON bool, args ...string) (st
 	var output, stderr string
 	var err error
 
-	// If Multus is enabled we proxy all the command to the mgr sidecar
-	if c.clusterInfo.NetworkSpec.IsMultus() {
-		output, stderr, err = c.Context.RemoteExecutor.ExecCommandInContainerWithFullOutputWithTimeout(c.clusterInfo.Context, cephclient.ProxyAppLabel, cephclient.CommandProxyInitContainerName, c.clusterInfo.Namespace, append([]string{"radosgw-admin"}, args...)...)
-	} else {
-		command, args := cephclient.FinalizeCephCommandArgs("radosgw-admin", c.clusterInfo, args, c.Context.ConfigDir)
-		output, err = c.Context.Executor.ExecuteCommandWithTimeout(exec.CephCommandsTimeout, command, args...)
-	}
+	// always proxy all the command to the mgr sidecar because radosgw-admin does not maintain
+	// forward compatibility. older rook operator versions that attempt to operate on newer ceph
+	// versions often see errors with the object store due to changes in radosgw-admin behavior
+	output, stderr, err = c.Context.RemoteExecutor.ExecCommandInContainerWithFullOutputWithTimeout(c.clusterInfo.Context, cephclient.ProxyAppLabel, cephclient.CommandProxySidecarContainerName, c.clusterInfo.Namespace, append([]string{"radosgw-admin"}, args...)...)
```

Comment:
This is quite a heavyweight operation to proxy every radosgw-admin command to the mgr pod. It's not a very common issue that there would be an incompatibility. Another idea is to first execute the command locally and only if it fails, then try again remotely. This way, the happy case will still get the advantage of avoiding the hop to the mgr pod.

Reply:
I'd like to think on it. And let's chat about it. I'm not totally opposed, but it's more complexity and more code paths that can go wrong overall, so I have hesitations.

Reply:
I think my biggest hesitation is how difficult that would make unit testing. To make a reconcile-level unit test to report an error from a `radosgw-admin` command, we will have to set up the environment so that it returns errors for both local and remote commands, or massage the environment such that it only runs one of the variants -- the easiest way would be to set the network spec to multus.

But I suppose that is an indication that we have problems with our frameworks and with not using enough dependency injection than it is with your suggestion, which is a good one.